newIDE/app/src/EventsSheet/InstructionEditor/TextEditorDialog.js

```diff
@@ -0,0 +1,301 @@
+            label={<Trans>Apply</Trans>}
+            primary
+            keyboardFocused
+            onClick={() => {
+              if (eventType === 'BuiltinCommonInstructions::Comment') {
+                this.setComment();
+              } else if (eventType === 'BuiltinCommonInstructions::Group') {
+                this.setName();
+              }
+              this.setTextColors();
+              this.setBackgroundColors();
+
+              onApply();
+            }}
+          />,
+        ]}
+        secondaryActions={[
+          <HelpButton
+            key="help"
+            helpPagePath="/interface/scene-editor/layers-and-cameras"
+          />,
+        ]}
+      >
+        <Column noMargin>
+          <MiniToolbar>
+            <MiniToolbarText>
+              <Trans />
+            </MiniToolbarText>
+            <MiniToolbarText>
+              <Trans>Background color:</Trans>
+            </MiniToolbarText>
+            <ColorPicker
+              style={styles.sizeTextField}
+              disableAlpha
+              color={{
+                r: this.state.backgroundColors.red,
+                g: this.state.backgroundColors.green,
+                b: this.state.backgroundColors.blue,
+                a: 255,
+              }}
+              onChangeComplete={color => {
+                this.setState({
+                  backgroundColors: {
+                    red: color.rgb.r,
+                    green: color.rgb.g,
+                    blue: color.rgb.b,
+                  },
+                });
+
+                this.forceUpdate();
+              }}
+            />
+
+            {eventType !== 'BuiltinCommonInstructions::Group' && (
+              <React.Fragment>
+                <MiniToolbarText>
+                  <Trans>Text color:</Trans>
+                </MiniToolbarText>
+                <ColorPicker
+                  style={styles.sizeTextField}
+                  disableAlpha
+                  color={{
+                    r: this.state.textColors.red,
+                    g: this.state.textColors.green,
+                    b: this.state.textColors.blue,
+                    a: 255,
+                  }}
+                  onChangeComplete={color => {
+                    this.setState({
+                      textColors: {
+                        red: color.rgb.r,
+                        green: color.rgb.g,
+                        blue: color.rgb.b,
+                      },
+                    });
+
+                    this.forceUpdate();
+                  }}
+                />
+              </React.Fragment>
+            )}
+          </MiniToolbar>
+          <Line noMargin>
+            <Column expand>
+              <Line>
+                <SemiControlledTextField
+                  commitOnBlur
+                  hintText={t`Enter the text to be displayed`}
+                  fullWidth
+                  multiLine
+                  rows={100}
+                  rowsMax={100}
```

Comment:
That's too much! :) Let's use 8 as in the text editor.

Reply:
I set 100 for having a large window, because if you put more text in the window, textfield doesn't get bigger because limited by 8 rows.
We need to reopen the editor.

There is also a problem with <minitoolbar> and <dialog>.
Minitoolbar should not be included in the scroll of the dialog.
Maybe we need to have a new props "header" for the dialog. And we pass to header the minitoolbar. 

This is also anoying in for "edit hitboxes" "edit points" "add an animation+"

Reply:
> Minitoolbar should not be included in the scroll of the dialog.
> Maybe we need to have a new props "header" for the dialog. And we pass to header the minitoolbar.

Yes, I think this could maybe be solved on a case by case basis with the `<ScrollView>` component too (I'm a bit worried of modifying Dialog, I like to keep it simple) (though it's always possible to also make a component DialogWithHeader that uses ScrollView for the middle part). There is something to do.

> I set 100 for having a large window, because if you put more text in the window, textfield doesn't get bigger because limited by 8 rows.
We need to reopen the editor.

Can you set rows={a small number} and rowsMax={a higher number, like 30}? So that the field is taking more space but not too much.